models/issue.go

```diff
@@ -1104,21 +1104,10 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 	}
 
 	// Milestone and assignee validation should happen before insert actual object.
-
-	// There's no good way to identify a duplicate key error in database/sql; brute force some retries
-	dupIndexAttempts := issueMaxDupIndexAttempts
-	for {
-		_, err := e.SetExpr("`index`", "coalesce(MAX(`index`),0)+1").
-			Where("repo_id=?", opts.Issue.RepoID).
-			Insert(opts.Issue)
-		if err == nil {
-			break
-		}
-
-		dupIndexAttempts--
-		if dupIndexAttempts <= 0 {
-			return err
-		}
+	if _, err := e.SetExpr("`index`", "coalesce(MAX(`index`),0)+1").
+		Where("repo_id=?", opts.Issue.RepoID).
+		Insert(opts.Issue); err != nil {
+		return err
```

Comment:
Shouldn't err be wrap around ErrNewIssueInsert ?
```suggestion
		return ErrNewIssueInsert{err}
```

Reply:
Oh, man! Gross oversight on my part. Sorry. That's what happens when you move code around without using git.
Give me a minute; I'll check this properly (I need to artificially produce an error for that).

Reply:
@sapk fixed, checked and tested. I've used a trigger to produce the fail artificially:
```
CREATE OR REPLACE FUNCTION make_insert_fail()
  RETURNS trigger AS 
$BODY$
DECLARE
    is_longer_than_max_length BOOLEAN;
BEGIN
    RAISE EXCEPTION 'Error produced by BEFORE INSERT TRIGGER insert_fail.' USING ERRCODE='20808';
END
$BODY$
  LANGUAGE 'plpgsql' SECURITY INVOKER
;

CREATE TRIGGER insert_fail BEFORE INSERT ON issue
FOR EACH ROW EXECUTE PROCEDURE make_insert_fail();
```

Then, when attempting to create an issue, I've got in the log:
```
2019/10/01 13:25:47 models/issue.go:1205:NewIssue() [E] NewIssue: error attempting to insert the new issue; will retry. Original error: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
2019/10/01 13:25:47 models/issue.go:1205:NewIssue() [E] NewIssue: error attempting to insert the new issue; will retry. Original error: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
2019/10/01 13:25:47 ...outers/repo/issue.go:580:NewIssuePost() [E] NewIssue: NewIssue: too many errors attempting to insert the new issue. Last error was: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
```

For a pull request:
```
2019/10/01 13:29:07 models/pull.go:672:NewPullRequest() [E] NewPullRequest: error attempting to insert the new issue; will retry. Original error: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
2019/10/01 13:29:07 models/pull.go:672:NewPullRequest() [E] NewPullRequest: error attempting to insert the new issue; will retry. Original error: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
2019/10/01 13:29:07 routers/repo/pull.go:796:CompareAndPullRequestPost() [E] NewPullRequest: NewPullRequest: too many errors attempting to insert the new issue. Last error was: pq: Error produced by BEFORE INSERT TRIGGER insert_fail.
```

On the screen:

![image](https://user-images.githubusercontent.com/18600385/65981708-0e4db100-e450-11e9-9cf6-8e17b8daf183.png)

**Everything worked correctly** when I disabled the trigger:

```
alter table issue disable trigger insert_fail;
```